File: solutions/php/hamming/1/Hamming.php
<?php

declare(strict_types=1);

function distance(string $strandA, string $strandB): int
{
    // Check if strands are of equal length
    if (strlen($strandA) !== strlen($strandB)) {
        throw new InvalidArgumentException('DNA strands must be of equal length.');
    }
    
    $distance = 0;
    $length = strlen($strandA);
    
    // Compare each character position
    for ($i = 0; $i < $length; $i++) {
        if ($strandA[$i] !== $strandB[$i]) {
            $distance++;
        }
    }
    
    return $distance;
}

File: solutions/vbnet/hamming/1/Hamming.vb
Public Module Hamming
    Public Function Distance(ByVal firstStrand As String, ByVal secondStrand As String) As Integer
        ' Check if the strands have the same length
        If firstStrand.Length <> secondStrand.Length Then
            Throw New ArgumentException("Strands must be of equal length")
        End If
        
        Dim differences As Integer = 0
        
        ' Compare each position
        For i As Integer = 0 To firstStrand.Length - 1
            If firstStrand(i) <> secondStrand(i) Then
                differences += 1
            End If
        Next
        
        Return differences
    End Function
End Module